common/middleware/ensure-authenticated.js

```diff
@@ -9,13 +9,33 @@ function _isExpired(authExpiry) {
 module.exports = function ensureAuthenticated({
   provider,
   whitelist = [],
+  expiryMargin = 5 * 60,
 } = {}) {
   return (req, res, next) => {
-    if (whitelist.includes(req.url) || !_isExpired(req.session.authExpiry)) {
+    let authExpiry = req.session.authExpiry
+    if (req.method === 'GET') {
+      authExpiry -= expiryMargin
+    }
+    if (whitelist.includes(req.url) || !_isExpired(authExpiry)) {
       return next()
     }
 
     req.session.originalRequestUrl = req.originalUrl
+    if (req.method === 'POST') {
+      let errorKey
+      const contentType = req.header('content-type')
+      if (contentType.startsWith('multipart/form-data')) {
+        errorKey = 'MULTIPART_FAILED_AUTH'
+      } else if (req.xhr) {
+        errorKey = 'DELETE_FAILED_AUTH'
```

Comment:
This also feels a bit presumptuous and dangerous to say that an XHR request is always going to be to do with documents failing to delete.

I kind of feel this error handling needs to be kept closer to documents and the form wizard rather than tied to the whole app level authentication.

Reply:
It is presumptuous, but until the auth token duration is sorted out and this can be handled properly, I thought it was premature optimisation to do more than this.

Reply:
Sorted